manager/role_manager_test.go

```diff
@@ -0,0 +1,213 @@
+package manager
```

Comment:
How do we test the specific condition which this fix handles: start with a raft node in the memberlist but not in the store and make sure that role manager removes it when it starts up. @cyli

Reply:
https://github.com/docker/swarmkit/pull/2551/files#diff-5ebf7b379f77f441db1a9548cabf2a89R170:

Because the role manager is being independently tested, so it can be started at any time.  I create a 3-node raft cluster, and update the store to reflect only 2 node objects.  The membership should reflect 3 raft nodes, but the store says that there are 2 canonical node objects.  Then I create a new role manager using one of the raft nodes, and I test that after a while one of the nodes has been kicked out.

I'll add another check that the correct raft node has been kicked out.

Reply:
Huh, maybe we can do the same thing for the control api server.